joint_trajectory_controller/test/test_trajectory_controller.cpp

```diff
@@ -548,6 +548,40 @@ TEST_P(TrajectoryControllerTestParameterized, state_topic_consistency)
   }
 }
 
+/**
+ * @brief check if hold on startup is deactivated
+*/
+TEST_P(TrajectoryControllerTestParameterized, no_hold_on_startup)
+{
+  rclcpp::executors::MultiThreadedExecutor executor;
+
+  rclcpp::Parameter start_holding_parameter("start_holding", false);
+  SetUpAndActivateTrajectoryController(executor, true, {start_holding_parameter});
+
+  constexpr auto FIRST_POINT_TIME = std::chrono::milliseconds(250);
+  updateController(rclcpp::Duration(FIRST_POINT_TIME));
+  ASSERT_FALSE(traj_controller_->carryingOutTrajectory());
+
+  executor.cancel();
+}
+
+/**
+ * @brief check if hold on startup
+*/
+TEST_P(TrajectoryControllerTestParameterized, hold_on_startup)
+{
+  rclcpp::executors::MultiThreadedExecutor executor;
+
+  rclcpp::Parameter start_holding_parameter("start_holding", true);
+  SetUpAndActivateTrajectoryController(executor, true, {start_holding_parameter});
+
+  constexpr auto FIRST_POINT_TIME = std::chrono::milliseconds(250);
+  updateController(rclcpp::Duration(FIRST_POINT_TIME));
+  ASSERT_TRUE(traj_controller_->carryingOutTrajectory());
```

Comment:
Can you add more comments to better understand the tests.

Reply:
I added comments now, is this clearer?
Now I see that tests are missing if it is really holding after a call of `set_hold_position`. But this should be an easy one.